shotover/src/transforms/query_counter.rs

```diff
@@ -26,11 +29,25 @@ pub struct QueryCounterConfig {
 
 impl QueryCounter {
     pub fn new(counter_name: String) -> Self {
-        counter!("shotover_query_count", "name" => counter_name.clone());
+        let counter_name_ref: &'static str = counter_name.leak();
+        let mut counters = HashMap::new();
+        let query_counter = counter!("shotover_query_count", "name" => counter_name_ref);
+        counters.insert("shotover_query_count".to_string(), query_counter);
 
         QueryCounter {
             // Leaking here is fine since the builder is created only once during shotover startup.
-            counter_name: counter_name.leak(),
+            counter_name: counter_name_ref,
+            counters,
+        }
+    }
+
+    fn increment_counter(&mut self, query: String, query_type: &'static str) {
+        if self.counters.contains_key(query.as_str()) {
```

Comment:
The hashmap provides an API to:
*  fetch the value if it exists
* otherwise reuse create, insert and return a value

It used by calling [entry](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry) and then calling [https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with](Entry::or_insert_with)
Which will look like:
```rust
self.counters
  .entry(query.as_str())
  .or_insert_with(|| counter!("shotover_query_count", "name" => self.counter_name, "query" => query.clone(), "type" => query_type))
  .increment(1);
```

This should be:
* simpler - its now a one liner
* faster - we only have to perform the hashmap lookup once

Reply:
This is handy!